src/devices/cpu/drcbearm64.h

```diff
@@ -0,0 +1,286 @@
+// license:BSD-3-Clause
+// copyright-holders:windyfairy
+#ifndef MAME_CPU_DRCBEARM64_H
+#define MAME_CPU_DRCBEARM64_H
+
+#pragma once
+
+#include "drcuml.h"
+#include "drcbeut.h"
+
+#include "asmjit/src/asmjit/asmjit.h"
+#include "asmjit/src/asmjit/a64.h"
+
+#include <vector>
+
+
+namespace drc {
+
+class drcbe_arm64 : public drcbe_interface
+{
+	typedef uint32_t (*arm64_entry_point_func)(void *entry);
+
+public:
+	drcbe_arm64(drcuml_state &drcuml, device_t &device, drc_cache &cache, uint32_t flags, int modes, int addrbits, int ignorebits);
+	virtual ~drcbe_arm64();
+
+	virtual void reset() override;
+	virtual int execute(uml::code_handle &entry) override;
+	virtual void generate(drcuml_block &block, const uml::instruction *instlist, uint32_t numinst) override;
+	virtual bool hash_exists(uint32_t mode, uint32_t pc) override;
+	virtual void get_info(drcbe_info &info) override;
+	virtual bool logging() const override { return false; }
+
+private:
+	class be_parameter
+	{
+		static int const REG_MAX = 30;
```

Comment:
Better to use `static inline constexpr` for these to avoid weird linker errors if you do something that takes a reference to it.